pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -864,10 +902,28 @@ public Map<String, Map<String, String>> getControllerJobs(
     Set<String> jobTypesToFilter = null;
     if (StringUtils.isNotEmpty(jobTypesString)) {
       jobTypesToFilter = new HashSet<>(java.util.Arrays.asList(StringUtils.split(jobTypesString, ',')));
+    }
+     // Validate inputs
+    Set<String> validJobTypes =
+            java.util.Arrays.stream(ControllerJobType.values()).map(ControllerJobType::name)
+                    .collect(Collectors.toSet());
+    if (jobTypesToFilter != null) {
+      for (String jobTypeFilterInput : jobTypesToFilter) {
+        if (!validJobTypes.contains(jobTypeFilterInput)) {
+          throw new IllegalArgumentException("Valid jobTypes are: " + validJobTypes);
+        }
+      }
+      if (jobTypesToFilter.contains(ControllerJobType.RELOAD_ALL_SEGMENTS.name())) {
+        // We want to get rid of RELOAD_ALL_SEGMENTS given both the reload types map to
+        // the same jobType RELOAD_SEGMENT . This is to be backwards compatible until we've removed
+        // all uses (eg: UI) of RELOAD_ALL_SEGMENTS
```

Comment:
I think that should only be this, https://github.com/apache/pinot/blob/master/pinot-controller/src/main/resources/app/pages/TenantDetails.tsx#L387

Reply:
Thanks. Updated the PR.